packages/astro/src/core/middleware/loadMiddleware.ts

```diff
@@ -8,9 +10,10 @@ import { MIDDLEWARE_MODULE_ID } from './vite-plugin.js';
  */
 export async function loadMiddleware(moduleLoader: ModuleLoader) {
 	try {
-		const module = await moduleLoader.import(MIDDLEWARE_MODULE_ID);
-		return module;
-	} catch {
-		return void 0;
+		return await moduleLoader.import(MIDDLEWARE_MODULE_ID);
+	} catch (error: any) {
+		const astroError = new AstroError(MiddlewareCantBeLoaded);
+		astroError.cause = error.cause;
```

Comment:
```suggestion
		const astroError = new AstroError(MiddlewareCantBeLoaded, { cause: error });
```
We missed the actual error here.

Reply:
Weird, I couldn’t find an example like this

Reply:
Actually, worth looking whether AstroError passes the cause to `super()`.

Reply:
It does, but in a roundabout way.
```suggestion
		const astroError = new AstroError(MiddlewareCantBeLoaded, undefined, { cause: error });
```